File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

resistor_band_t* colors() {
    static resistor_band_t color[TYPES];

    for(int i = BLACK; i < TYPES; ++i){
        color[i] = i;
    }

    return color;
}

int color_code(resistor_band_t code) {
  return (int) code;
}

File: elixir/resistor-color/lib/resistor_color.ex
defmodule ResistorColor do
  @moduledoc """
  Helping with color-coded resistors.
  """

  @colors ~w(black brown red orange yellow green blue violet grey white)a

  @doc """
  Return all colors as a list.
  """
  @spec colors() :: list(String.t())
  def colors(), do: @colors

  @doc """
  Return the value of a color band.
  """
  @spec code(atom()) :: non_neg_integer()
  def code(color), do: Enum.find_index(@colors, fn c -> c == color end)
end
